scripts/test-uplink-share.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+TMPDIR=$(mktemp -d -t tmp.XXXXXXXXXX)
+
+cleanup(){
+    rm -rf "$TMPDIR"
+    uplink --access "$GATEWAY_0_ACCESS" rm "sj://$BUCKET_WITH_ACCESS/$FOLDER_TO_SHARE_FILE/testfile"
+    uplink --access "$GATEWAY_0_ACCESS" rm "sj://$BUCKET_WITH_ACCESS/another-testfile"
+    uplink --access "$GATEWAY_0_ACCESS" rm "sj://$BUCKET_WITHOUT_ACCESS/another-testfile"
+    uplink --access "$GATEWAY_0_ACCESS" rb "sj://$BUCKET_WITHOUT_ACCESS"
+    uplink --access "$GATEWAY_0_ACCESS" rb "sj://$BUCKET_WITH_ACCESS"
+    echo "cleaned up test successfully"
+}
+trap cleanup EXIT
+
+BUCKET_WITHOUT_ACCESS=bucket1
+BUCKET_WITH_ACCESS=bucket2
+
+FOLDER_TO_SHARE_FILE=folder
+
+SRC_DIR=$TMPDIR/source
+DST_DIR=$TMPDIR/dst
+
+mkdir -p "$SRC_DIR" "$DST_DIR"
+
+random_bytes_file () {
+    size=$1
+    output=$2
+    head -c $size </dev/urandom > $output
+}
+
+random_bytes_file "2048"       "$SRC_DIR/another-testfile"  # create 2kb file of random bytes (inline)
+random_bytes_file "5120"       "$SRC_DIR/testfile"          # create 5kb file of random bytes (inline)
+uplink --access "$GATEWAY_0_ACCESS" mb "sj://$BUCKET_WITHOUT_ACCESS/"
+uplink --access "$GATEWAY_0_ACCESS" mb "sj://$BUCKET_WITH_ACCESS/"
+
+uplink --access "$GATEWAY_0_ACCESS" cp "$SRC_DIR/testfile" "sj://$BUCKET_WITH_ACCESS/$FOLDER_TO_SHARE_FILE/"
+uplink --access "$GATEWAY_0_ACCESS" cp "$SRC_DIR/another-testfile" "sj://$BUCKET_WITH_ACCESS/"
+uplink --access "$GATEWAY_0_ACCESS" cp "$SRC_DIR/another-testfile" "sj://$BUCKET_WITHOUT_ACCESS/"
+
+# Make access with readonly rights
+SHARED_ACCESS=$(uplink --access "$GATEWAY_0_ACCESS" share --allowed-path-prefix sj://$BUCKET_WITH_ACCESS/$FOLDER_TO_SHARE_FILE/ --readonly | grep Access | cut -d: -f2)
+
+check_exit_code(){
+	if [ $1 -eq 0 ]; then
+    	echo "Result of copying does not match expectations. Test FAILED"
+    	exit 1
+	else
+		echo "Copy file without permission: PASSED"    # Expect unsuccessful exit code
+	fi
+}
+
+uplink cp "$SRC_DIR/another-testfile" "sj://$BUCKET_WITH_ACCESS/$FOLDER_TO_SHARE_FILE/" --access $SHARED_ACCESS
```

Comment:
is this upload failing because `--readonly` in `share` command? mainly I'm thinking aloud about what we are testing here.

Reply:
Yes, it is failing and we expect this. We check if `--readonly` flag works

Reply:
I'm only concerned that we don't have a check that will fail if upload will be successful (e.g. because of some regression). We are only checking exit code and I think we will return 0 in both cases.

Reply:
In this case, we have exit code 1 so if the upload will be successful we will have 0 and test will fail.

Reply:
I think I confused you in the previous answer ^

Reply:
ah I just realized that `check_exit_code` is doing something else I was thinking :) can we rename it to something like `require_error_exit_code` or `require_error_code`? it will match pattern we have in Go and C tests. Will be easy to understand for our devs.

Reply:
Done